### ISSUE TITLE
bug in parallel method?

### ISSUE THREAD

**Issue by komelgman** [cho45/jsdeferred#13]
``` javascript
// in this code "d" may be not Deferred
str 456: if (typeof d == "function") d = Deferred.next(d); 

// in this code "d" always Deferred
str 474 :       for (var i in dl) if (dl.hasOwnProperty(i)) {
str 475 :           dl[i].cancel();
str 476 :       }
```

is a bug or I do not understand the code correctly?

``` javascript
Deferred.parallel = function (dl) {
    if (arguments.length > 1) dl = Array.prototype.slice.call(arguments);
    var ret = new Deferred(), values = {}, num = 0;
    for (var i in dl) if (dl.hasOwnProperty(i)) (function (d, i) {
        if (typeof d == "function") d = Deferred.next(d);
        d.next(function (v) {
            values[i] = v;
            if (--num <= 0) {
                if (dl instanceof Array) {
                    values.length = dl.length;
                    values = Array.prototype.slice.call(values, 0);
                }
                ret.call(values);
            }
        }).error(function (e) {
            ret.fail(e);
        });
        num++;
    })(dl[i], i);

    if (!num) Deferred.next(function () { ret.call() });
    ret.canceller = function () {
        for (var i in dl) if (dl.hasOwnProperty(i)) {
            dl[i].cancel();
        }
    };
    return ret;
};
```
